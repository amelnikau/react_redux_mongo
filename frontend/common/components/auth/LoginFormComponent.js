import React from 'react'
import {connect} from 'react-redux'
import * as actions from '../../actions/index'
import { withRouter } from 'react-router-dom';

let LoginFormComponent = ({setErrorMessage, login, signUp, history}) => {
    let passwordInputValue;
    let usernameInputValue;

    function handleLoginActivity(e, action) {
        e.preventDefault();
        if (hasErrors()) return;
        action(usernameInputValue.value, passwordInputValue.value, history);
    }

    function onLoginClick(e) {
       handleLoginActivity(e, login);
    }

    function onSignUpClick(e) {
        handleLoginActivity(e, signUp);
    }

    function hasErrors() {
        if (!usernameInputValue.value.trim() || !passwordInputValue.value.trim()) {
            setErrorMessage("Required field value missing");
            return true;
        }
        return false;
    }

    return (
        <div>
            Username: <input type='text' ref={node => {usernameInputValue = node}}/><br/>
            Password: <input type='password' ref={node => {passwordInputValue = node}}/><br/>
            <input type='button' onClick={onLoginClick} value='Log In'/>
            <input type='button' onClick={onSignUpClick} value='Sign Up'/>
        </div>
    )
};

LoginFormComponent = withRouter(connect(undefined, actions)(LoginFormComponent));

export default LoginFormComponent