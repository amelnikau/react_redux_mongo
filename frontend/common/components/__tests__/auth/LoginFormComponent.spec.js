import LoginFormComponent from '../../../components/auth/LoginFormComponent';
import renderer from 'react-test-renderer';
import React from 'react'
import configureMockStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import {Route} from 'react-router-dom'
import BrowserRouter from 'react-router-dom/BrowserRouter';

const mockStore = configureMockStore();

describe('Snapshot components testing', () => {
    describe('LoginFormComponent', () => {
        it("should render login form", () => {
            const tree = renderer.create(
                <Provider
                    store={mockStore({})}><BrowserRouter><Route><LoginFormComponent/></Route></BrowserRouter></Provider>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    })
});