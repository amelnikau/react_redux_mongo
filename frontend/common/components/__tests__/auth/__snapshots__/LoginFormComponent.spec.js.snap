// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot components testing LoginFormComponent should render login form 1`] = `
<div>
  Username: 
  <input
    type="text"
  />
  <br />
  Password: 
  <input
    type="password"
  />
  <br />
  <input
    onClick={[Function]}
    type="button"
    value="Log In"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Sign Up"
  />
</div>
`;
